core/src/test/scala/pureconfig/CoproductConvertersSuite.scala

```diff
@@ -21,6 +21,12 @@ class CoproductConvertersSuite extends BaseSuite {
     ConfigConvert[AnimalConfig].from(conf.root()) shouldEqual Right(DogConfig(2))
   }
 
+  it should "read disambiguation information on sealed families with the cases nested in the companion" in {
+    import CarMaker._
+    val conf = ConfigFactory.parseString("{ type = bmw }")
+    ConfigConvert[CarMaker].from(conf.root()) shouldEqual Right(BMW)
```

Comment:
Strange bug, I was expecting this to work because `type` is set and correct and no other arguments are needed to build `BMW`. @ruippeixotog do you know what's this about?